app/dose-event-fab/dose-event-fab.component.ts

```diff
@@ -15,11 +16,20 @@ export class DoseEventFabComponent {
     @Input()
     public doseEvents: Array<DoseEvent>;
 
+
+    constructor(
+        private doseEventService: DoseEventService
+    ) {
+    }
+
     takeAll() {
         for (let dose of this.doseEvents) {
             dose.action = "TAKEN";
         }
-
+        /* Ignore server response, just assume it worked. */
+        this.doseEventService.saveList(
+            this.doseEvents
+        );
     }
```

Comment:
Clicking the take all FAB on the Dose Events screen now saves that all were taken to the server.

I am not doing anything with the promise right now, just assuming it worked.